src/Impl/Widget.cpp

```diff
@@ -0,0 +1,98 @@
+#include "Widget.h"
+
+#include "Plug/Event.h"
+#include "Plug/Graphics.h"
+#include "Plug/LayoutBox.h"
+
+Widget::Widget(const plug::LayoutBox& box) : box_(box.clone()) {}
+
+Widget::Widget(const Widget& widget) : box_(nullptr) { operator=(widget); }
+
+Widget& Widget::operator=(const Widget& widget)
+{
+  delete box_;
+  box_ = widget.box_->clone();
+  return *this;
+}
```

Comment:
`delete nullptr` is undefined behavior and will likely cause program to crash. Do not implement copy-constructor using `operator=`